mtt/src/mtt_types.rs

```diff
@@ -118,3 +116,226 @@ impl MttL1EntryType {
         }
     }
 }
+
+// Bits 45:26
+const L2_ADDR_MASK: u64 = 0x3fff_fc00_0000;
+// Shifted right by 26, then 3-bit left shift to account for the 8-byte entry
+const L2_ADDR_SHIFT: u64 = 23;
+
+fn get_l2_entry(phys_addr: SupervisorPageAddr) -> MttL2EntryType {
+    let addr = phys_addr.bits();
+    let mtt_index = (addr & L2_ADDR_MASK) >> L2_ADDR_SHIFT;
+    let mtt_entry_addr = (get_mtt_base() + mtt_index) as *const u64;
+    assert!(mtt_index <= (8 * 1024 * 1024) - 8);
```

Comment:
Can this fail? If so, should return an error here and should be checked before it's used.
what is the magic 8MB-8 number?

Reply:
Ack. The assert was checking that we can't possibly exceed 8MB (the last index is 8 bytes less since we write 8-bytes at at time).

Reply:
The asserts have been removed. It's actually impossible to exceed the 8MB range by construction, so it's redundant anyway.